src/ipc/ipc.c

```diff
@@ -471,9 +472,12 @@ int ipc_init(struct sof *sof)
 	list_init(&sof->ipc->shared_ctx->msg_list);
 	list_init(&sof->ipc->shared_ctx->comp_list);
 
-	for (i = 0; i < MSG_QUEUE_SIZE; i++)
-		list_item_prepend(&sof->ipc->shared_ctx->message[i].list,
+	for (i = 0; i < MSG_QUEUE_SIZE; i++) {
```

Comment:
@tlauda What's the benefit if we anyways allocate MSG_QUEUE_SIZE of messages here? Earlier it's in rzalloc() of shared_ctx, now here...

Reply:
@kv2019i One of the benefits is getting them cache line aligned for free, but mostly it's a preparation for further multicore based changes. I'm planning on adding separate heap for shared memory, providing separate sections for shared static data and also extracting "uncached" related code to specific platforms.

Reply:
@tlauda About the uncached... I'm thinking about the fact that the imx platform doesn't have uncached memory (save for the actual IO-mapped devices) so the platform specific macros are lying on whether the memory is cached or not (not sure what lie they tell right now)

Reply:
@tlauda Ack ok. I got misled a bit with the PR title "optimize memory usage", but the commit message for this patch is actually more accurate. Others seem to agree, so good for me as well.